File: solutions/643. Maximum Average Subarray I/643.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Float}
def find_max_average(nums, k)
    return nums[0] if nums.size == 1 
    max_average = -1 * Float::INFINITY
    total_nums = 0
    nums.each_with_index do |num, i|
        if i >= k
            max_average = [max_average, total_nums/k.to_f].max
            total_nums -= nums[i-k]
        end
        total_nums += nums[i]
    end
    [max_average, total_nums/k.to_f].max
end

File: solutions/643. Maximum Average Subarray I/643.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        if len(nums) == 1: return nums[0]
        max_average = float('-inf')
        total_nums = 0
        for i in range(len(nums)):
            if i >= k:
                max_average = max(max_average, total_nums/k)
                total_nums -= nums[i-k]
            total_nums += nums[i]
        return max(max_average, total_nums/k)